azkaban-common/src/main/java/azkaban/imagemgmt/permission/PermissionManagerImpl.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.permission;
+
+import azkaban.imagemgmt.daos.ImageTypeDao;
+import azkaban.imagemgmt.exeception.ImageMgmtException;
+import azkaban.imagemgmt.models.ImageOwnership;
+import azkaban.imagemgmt.models.ImageOwnership.Role;
+import azkaban.imagemgmt.models.ImageType;
+import azkaban.user.Permission;
+import azkaban.user.Permission.Type;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.collections4.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class defines and manages the permission for accessing image management APIs. The
+ * permissions are defined on image type. The permissions are created using user role and access
+ * type. This class internally accesses image type cache to get image type metadata.
+ */
+@Singleton
+public class PermissionManagerImpl implements PermissionManager {
+
+  private static final Logger log = LoggerFactory.getLogger(PermissionManagerImpl.class);
+
+  private final ImageTypeDao imageTypeDao;
+
+  /**
+   * Permission for image type owner of role ADMIN. ADMIN owner has complete access to image
+   * management APIs including add/delete of new member as image type owner.
+   */
+  private static final Permission ADMIN_PERMISSION = new Permission(Type.ADMIN);
+  /**
+   * Permission for image type owner of role MEMBER. MEMBER owner has complete access to image
+   * management APIs except add/delete of new member as image type owner.
+   */
+  private static final Permission MEMBER_PERMISSION = new Permission(Type.CREATE, Type.GET,
```

Comment:
Permission is an existing class and it has Type definition and leveraging the same in the code. Yes Admin and Member are the roles image owner.  By Permission we can define the API access such as create, update etc.

Reply:
Can you please rename existing Type class to PermissionType? I am unclear about why you are defining 
ADMIN_PERMISSION and MEMBER_PERMISSION. Why don't you have PermissionTypes associated as instance variable for enum itself for role?

Reply:
I don't see the need for ADMIN_PERMISSION here since it matches the already existing ADMIN permission in Azkaban. Why not directly ask for ADMIN permission?